ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/show/ShowPartitionsOperation.java

```diff
@@ -54,12 +54,16 @@ public ShowPartitionsOperation(DDLOperationContext context, ShowPartitionsDesc d
   @Override
   public int execute() throws HiveException {
     Table tbl = context.getDb().getTable(desc.getTabName());
-    if (!tbl.isPartitioned()) {
-      throw new HiveException(ErrorMsg.TABLE_NOT_PARTITIONED, desc.getTabName());
+    boolean isPartitionedIcebergTable = isPartitionedIcebergTable(tbl);
+    if (!(tbl.isPartitioned() || isPartitionedIcebergTable)) {
```

Comment:
Why don't we extend tbl.isPartitioned() with iceberg support instead of extra checks?
for checking if the table is Iceberg, we could create a helper method in HiveTableOperations next to validateTableIsIceberg or maybe directly in Table (i.e. tbl.isIceberg())

Reply:
And also, there is an `IcebergTableUtil` class as well. It is in the `iceberg.mr.hive` package. I don't know if we can use it here.